test/object-store/sync/app.cpp

```diff
@@ -3305,11 +3308,22 @@ TEMPLATE_TEST_CASE("app: collections of links integration", "[sync][app][collect
             wait_for_num_outgoing_links_to_equal(r2, r2_source_objs.get(0), expected_coll_size);
             auto linked_objects = test_type.get_links(r2_source_objs.get(0));
             require_links_to_match_ids(linked_objects, remaining_dest_object_ids);
+
+            bool coll_cleared = false;
+            auto token = test_type.add_notification_callback(r1, r1_source_objs.get(0),
+                                                             [&coll_cleared](CollectionChangeSet c) {
+                                                                 coll_cleared = c.collection_was_cleared;
+                                                             });
+
             r2->begin_transaction();
             test_type.clear_collection(r2_source_objs.get(0));
             r2->commit_transaction();
             expected_coll_size = 0;
             REQUIRE(test_type.size_of_collection(r2_source_objs.get(0)) == expected_coll_size);
+            // none of the containers are actually triggering a notification. Is this a general bug or there is
```

Comment:
I am still not getting the notification, the collection does get cleared, but I am not getting notified. This is true for all the type of containers we are testing Lists, Sets and Dictionaries. I wonder if this is a problem with my code or a more general issue. @ironage